src/pages/events/EventForm.tsx

```diff
@@ -0,0 +1,129 @@
+import {Button, Card, CardContent, Container, Divider, Typography} from '@material-ui/core';
+import Editor, {monaco} from '@monaco-editor/react';
+import React, {useEffect, useRef} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {Event, JSONSchema} from '../../api/types';
+import {makeThemeSelector} from '../../selector/settingsSelector';
+import {makeJsonSchemaDefinitionsSelector} from '../../selector/systemSchemaSelector';
+import AxiosResponseViewer from '../common/components/AxiosResponseViewer';
+import {Alert, AlertTitle} from '@material-ui/lab';
+import {convertJsonSchemaToEditorValue} from '../../util/convertJsonSchemaToEditorValue';
+import {makeEventErrorSelector, makeEventResponseSelector} from '../../selector/eventSelector';
+import {clearEvent, executeEvent} from '../../action/eventCommands';
+
+interface EventPayloadFormProps {
+    event: Event;
+}
+
+let monacoInstance: any = null;
+monaco.init().then(instance => monacoInstance = instance);
+
+const EventForm = (props: EventPayloadFormProps) => {
+
+    const dispatch = useDispatch();
+    const theme = useSelector(makeThemeSelector());
+    const jsonSchemaDefinitions: Record<string, JSONSchema> | null = useSelector(makeJsonSchemaDefinitionsSelector());
+    const response = useSelector(makeEventResponseSelector());
+    const error = useSelector(makeEventErrorSelector());
+    const valueGetterRef = useRef();
+    const editorRef = useRef();
+
+    useEffect(() => {
+        if (!editorRef.current) {
+            return;
+        }
+
+        dispatch(clearEvent({}));
+        updateEditorModel();
+        /* eslint-disable-next-line react-hooks/exhaustive-deps */
+    }, [props.event]);
+
+    const handleEditorDidMount = (valueGetter: any, editor: any) => {
+        valueGetterRef.current = valueGetter;
+        editorRef.current = editor;
+
+        updateEditorModel();
+    };
+
+    const updateEditorModel = () => {
+        const defaultEditorValue: Record<string, any> = convertJsonSchemaToEditorValue(props.event.schema, jsonSchemaDefinitions || {});
+
+        const jsonCode = JSON.stringify(defaultEditorValue, null, 2);
+        const modelUri = monacoInstance.Uri.parse(`command_${props.event.eventName}.json`);
```

Comment:
I‘m not entirely sure but I believe that there is the possibility of commands and events having a conflict if they are named similarely. The Uri more or less identifies the document which means that similar named commands and events would have the same identifier

Reply:
e.g. https://github.com/event-engine/cockpit/blob/master/src/pages/queries/QueryPayloadForm.tsx#L52

Reply:
We should probably remove the code duplication regarding the monaco editors too. I can take care of that later though

Reply:
good catch, I'm fixing it

Reply:
Changed the prefix. Would be nice if you can take over the code duplication issue @martin-schilling